### PR TITLE
Fail elegantly when build script doesn't produce an output file

### PR DESCRIPTION
**Pull request by thaddeus**
Addresses #60

### CODE REVIEW

File: lib/moonshot/build_mechanism/script.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def build_hook(version)
 
   def post_build_hook(_version)
     unless File.exist?(@output_file) # rubocop:disable GuardClause
-      raise Thor::Error, 'Build command did not produce output file!'
+      step.failure 'Build command did not produce output file!'
```
Comment (glennpratt) [inline lib/moonshot/build_mechanism/script.rb]: Not sure this will communicate the error upstream (like raise would) allowing a non-zero exit.

Comment (thaddeus) [inline lib/moonshot/build_mechanism/script.rb]: Oops, you're right. I'm fixing the CF resources I'm using, then I'll test it out again.

Comment (glennpratt) [inline lib/moonshot/build_mechanism/script.rb]: This is how @jamesiarmes handled that.

https://github.com/acquia/moonshot/blob/master/lib/moonshot/controller.rb#L67-L78

Perhaps we want to consider how we can do that more generally with the hooks.

**Comment by askreet** [comment]
This PR is now incompatible with 1.0 -- please re-open if you have a chance to bring the fix up to date.

**Pull request closed without merging**